### PR TITLE
[ATL] Improve CAtlList, partially implement CAtlArray

### PR DESCRIPTION
**Pull request by learn-more**
Minimal implementations for just the parts that I need.

### CODE REVIEW

File: modules/rostests/apitests/atl/devenv/ATLTest.sln [hunk 1]
```diff
@@ -17,6 +17,10 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "CComQIPtr", "CComQIPtr.vcxp
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "CAtlFileMapping", "CAtlFile.vcxproj", "{3AE82A8E-D43D-41F6-8093-9C687283FAB6}"
 EndProject
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "CAtlArray", "CAtlArray.vcxproj", "{E3DB2078-6AE9-463E-830F-36C03FFE5DC3}"
```
Comment (katahiromz) [inline modules/rostests/apitests/atl/devenv/ATLTest.sln]: Modify the GUID.

File: sdk/lib/atl/atlcoll.h [hunk 2]
```diff
@@ -136,6 +145,247 @@ class CElementTraits :
...
+
+template<typename E, class ETraits>
+const E& CAtlArray<E, ETraits >::GetAt(size_t iElement) const
+{
+    ATLASSERT(iElement < m_Size);
+
+    return m_pData[iElement];
+}
+
+
```
Comment (katahiromz) [inline sdk/lib/atl/atlcoll.h]: Extra empty lines

File: sdk/lib/atl/atlcoll.h [hunk 3]
```diff
@@ -136,6 +145,247 @@ class CElementTraits :
...
+    }
+
+    return m_Size - 1;
+}
+
+
+template<typename E, class ETraits>
+bool CAtlArray<E, ETraits >::SetCount(size_t nNewSize, int nGrowBy /*= -1*/)
+{
+
```
Comment (katahiromz) [inline sdk/lib/atl/atlcoll.h]: empty line

File: modules/rostests/apitests/atl/CAtlList.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    CAtlList<int> list1;
+
+    ok(list1.GetCount() == 0, "Expected list1's size is zero, was %d\n", list1.GetCount());
+    list1.AddTail(56);
+    ok(list1.GetCount() == 1, "Expected list1's size is 1, was %d\n", list1.GetCount());
+    POSITION head = list1.AddHead(12);
+    ok(list1.GetCount() == 2, "Expected list1's size is 2, was %d\n", list1.GetCount());
+    POSITION tail = list1.AddTail(90);
+    ok(list1.GetCount() == 3, "Expected list1's size is 3, was %d\n", list1.GetCount());
+
```
Comment (katahiromz) [inline modules/rostests/apitests/atl/CAtlList.cpp]: empty line

File: sdk/lib/atl/atlcoll.h [hunk 5]
```diff
@@ -136,6 +145,247 @@ class CElementTraits :
...
+        CreateItems(m_pData + m_Size, nNewSize - m_Size);
+        m_Size = nNewSize;
+    }
+
+    return true;
+}
+
+template<typename E, class ETraits>
+size_t CAtlArray<E, ETraits >::GetCount() const
```
Comment (katahiromz) [inline sdk/lib/atl/atlcoll.h]: Why space after `ETraits`?

Comment (learn-more) [reply sdk/lib/atl/atlcoll.h]: I copied the template stuff from `CAtlList`

**Pull request merged**